app/angular/src/client/preview/angular-beta/StorybookModule.ts

```diff
@@ -61,22 +61,28 @@ export const getStorybookModuleMetadata = (
     props
   );
 
+  const isStandalone = isStandaloneComponent(component);
   // Look recursively (deep) if the component is not already declared by an import module
   const requiresComponentDeclaration =
     isDeclarable(component) &&
     !isComponentAlreadyDeclaredInModules(
       component,
       moduleMetadata.declarations,
       moduleMetadata.imports
-    );
+    ) &&
+    !isStandalone;
 
   return {
     declarations: [
       ...(requiresComponentDeclaration ? [component] : []),
       ComponentToInject,
       ...(moduleMetadata.declarations ?? []),
     ],
-    imports: [BrowserModule, ...(moduleMetadata.imports ?? [])],
+    imports: [
+      BrowserModule,
+      ...(isStandalone ? [component] : []),
```

Comment:
I you want test this you can add a test here : https://github.com/storybookjs/storybook/blob/41752fe5bc8ed2831a21b96bc5203b53c0e9b123/app/angular/src/client/preview/angular-beta/RendererFactory.test.ts